src/cluster_legacy.c

```diff
@@ -2610,6 +2614,9 @@ uint32_t writePingExt(clusterMsg *hdr, int gossipcount)  {
 /* We previously validated the extensions, so this function just needs to
  * handle the extensions. */
 void clusterProcessPingExtensions(clusterMsg *hdr, clusterLink *link) {
+    if (!server.process_clustermsg_extensions) {
```

Comment:
This is a bit hacky IMO. Rather, I would suggest pulling L2617 (or2624) up and bailing out of this function if `extensions == 0`, as a more generic fix.